lib/src/idx/trees/btree.rs

```diff
@@ -449,67 +531,164 @@ where
 	async fn deleted_from_internal(
 		&mut self,
 		tx: &mut Transaction,
-		store: &mut BTreeNodeStore<BK>,
+		store: &mut BTreeStore<BK>,
 		keys: &mut BK,
 		children: &mut Vec<NodeId>,
 		key_to_delete: Key,
 	) -> Result<(bool, Key, NodeId), Error> {
+		#[cfg(debug_assertions)]
+		debug!(
+			"Delete from internal - key_to_delete: {} - keys: {keys}",
+			String::from_utf8_lossy(&key_to_delete)
+		);
 		let left_idx = keys.get_child_idx(&key_to_delete);
 		let left_id = children[left_idx];
-		let mut left_node = store.get_node(tx, left_id).await?;
+		let mut left_node = store.get_node_mut(tx, left_id).await?;
+		// if the child y that precedes k in nodexx has at least t keys
 		if left_node.n.keys().len() >= self.state.minimum_degree {
 			// CLRS: 2a -> left_node is named `y` in the book
-			if let Some((key_prim, payload_prim)) = left_node.n.keys().get_last_key() {
-				keys.remove(&key_to_delete);
-				keys.insert(key_prim.clone(), payload_prim);
-				store.set_node(left_node, true)?;
-				return Ok((false, key_prim, left_id));
+			#[cfg(debug_assertions)]
+			debug!(
+				"CLRS: 2a - key_to_delete: {} - left: {left_node} - keys: {keys}",
+				String::from_utf8_lossy(&key_to_delete)
+			);
+			let (key_prim, payload_prim) = self.find_highest(tx, store, left_node).await?;
```

Comment:
That was part 1 of the bug. Previously I took the preceding value in the same node, which is wrong. I had to take the preceding value in the subtree. That's what `find_highest` is doing.